crates/swc_common/src/errors/mod.rs

```diff
@@ -874,6 +874,10 @@ impl Handler {
             }
         }
     }
+
+    pub fn take_diagnostics(&self) -> Vec<String> {
```

Comment:
This sounds good. Nice work!

Reply:
Great, you can merge it at any time.
> Can we have an option to set the formatting of error messages?

About `option to set the formatting of error messages`, i can implement in another PR.